### ISSUE TITLE
V122-2 : Time Average broken, missing custom Mix Sensor upon restart

### ISSUE THREAD

**Issue by Dark4Codrutz** [Rem0o/FanControl.Releases#1023]
**Describe the bug**

I managed to update to v122 after #1019 without breaking error but this time I'm met with 'missing sensor error' in a custom sensor setup.

![image](https://user-images.githubusercontent.com/18706363/185583263-53bd6875-0ec7-4bb2-8a5d-832e255ee77a.png)

My custom sensors are configured like this:
* _constant_ : sensor file with fixed value
* _compute_ : Mix that subtracts temp A from _constant_
* _smooth_ : time average for _compute_
* _Calibration_ : Mix that averages _smooth_ with temp B
* _PECI Calibrated_ : Mix that averages _Calibration_ with temp C

The error /warning refers to _smooth_ no finding _compute_ value. No additional log information.

```
      {
        "Identifier": "Mix/PECI Calibrated",
        "Name": "PECI Calibrated",
        "SelectedMixFunction": 0,
        "SelectedSensors": [
          {
            "Identifier": "Mix/Calibration",
            "Name": "Calibration",
            "SelectedMixFunction": 0,
            "SelectedSensors": [
              {
                "Identifier": "Time/smooth",
                "Name": "smooth",
                "SelectedTempSource": {
                  "Identifier": "Mix/compute",
                  "Name": "compute",
                  "SelectedMixFunction": 4,
                  "SelectedSensors": [
                    {
                      "FileName": "constant",
                      "Identifier": "File/constant",
                      "Name": "constant"
                    },
                    {
                      "Identifier": "/lpc/nct6791d/temperature/3",
                      "IsHidden": false,
                      "Name": "Temperature #3",
                      "NickName": "Temperature #3"
                    }
                  ]
                },
                "SelectedTime": 5
              },
              {
                "Identifier": "/lpc/nct6791d/temperature/1",
                "IsHidden": false,
                "Name": "Temperature #1",
                "NickName": "CPU Socket"
              }
            ]
          },
          {
            "Identifier": "/lpc/nct6791d/temperature/0",
            "IsHidden": false,
            "Name": "CPU Core",
            "NickName": "CPU Core"
          }
        ]
      },
```

**Comment by Dark4Codrutz** [comment]
After tinkering with my custom sensors a bit I think I found the root of the issue:
If I do a Time Average of any kind of Custom Mix Sensor it will warn that the sensor is not found upon startup.
Time Average works fine with normal hw sensors.

**Comment by Dark4Codrutz** [comment]
Probably slightly unrelated :
The new update also doesn't remember 'Tray icon temperature source' if I select a Custom Mix Sensor.

**Comment by Rem0o** [comment]
See See V123.

**Comment by Dark4Codrutz** [comment]
> See V123.

```


19/08/2022 16:27:45: System.NullReferenceException: Object reference not set to an instance of an object.
   at FanControl.Domain.FanCurves.AutoFanCurve..ctor(IAutoFanCurveConfig autoFanCurve, ISensorRepository sensorRepository)
   at FanControl.ViewModels.AutoFanCurveViewModel.UpdateFanCurve()
   at FanControl.MVVM.FromCollectionSetter.Set[T](Action`1 setter, T value, String memberName)
   at FanControl.MVVM.FromCollectionSetter.SetProperty[T,U](Action`1 setter, U value, ObservableCollection`1 observableCollection, Func`2 getComparisonProperty, String memberName)
   at FanControl.ViewModels.AutoFanCurveViewModel.set_SelectedTempSource(ISensorSourceViewModel value)
   at FanControl.ViewModels.AutoFanCurveViewModel..ctor(IAutoFanCurveConfig autoFanCurve, Func`2 validateName, Action`1 deleteFanCurve, ObservableCollection`1 tempSources, IApplicationClock applicationClock, IDialogService dialogService)
   at FanControl.ViewModels.AutoFanCurveViewModel.Deserialize(IAutoFanCurveConfig autoFanCurve, Func`2 validateName, Action`1 deleteFanCurve, ObservableCollection`1 tempSources, IApplicationClock applicationClock, IDialogService dialogService)
   at FanControl.App.<>c__DisplayClass24_0.<RegisterTypes>b__13(MainWindowViewModel c, SerializableAutoFanCurve fc)
   at FanControl.ViewModels.FanCurveProvider`1.<Deserialize>d__4.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at FanControl.ViewModels.MainWindowViewModel.RestoreApplicationState(SerializableMainWindow state, IFanCurveProvider`1 fanCurveProvider, ICustomSensorProvider`1 customSensorProvider)
   at FanControl.ViewModels.MainWindowViewModel.RestoreApplicationState(SerializableMainWindow state)
```
Error loading config file, like v122-1 (not sure if exact log message)

**Comment by Rem0o** [comment]
Mind sharing your json config?

**Comment by Dark4Codrutz** [comment]
[Standard - Rework2 - Copy.json.txt](https://github.com/Rem0o/FanControl.Releases/files/9382974/Standard.-.Rework2.-.Copy.json.txt)
This works in v121
(ik I'm doing two averages when I can do just one, not the issue though)

**Comment by Dark4Codrutz** [comment]
> Mind sharing your json config?

Were you able to reproduce the issue ?

**Comment by Dark4Codrutz** [comment]
Maybe I should open a separate issue for this new error

> > See V123.
> 
> ```
> 
> 
> 19/08/2022 16:27:45: System.NullReferenceException: Object reference not set to an instance of an object.
>    at FanControl.Domain.FanCurves.AutoFanCurve..ctor(IAutoFanCurveConfig autoFanCurve, ISensorRepository sensorRepository)
>    at FanControl.ViewModels.AutoFanCurveViewModel.UpdateFanCurve()
>    at FanControl.MVVM.FromCollectionSetter.Set[T](Action`1 setter, T value, String memberName)
>    at FanControl.MVVM.FromCollectionSetter.SetProperty[T,U](Action`1 setter, U value, ObservableCollection`1 observableCollection, Func`2 getComparisonProperty, String memberName)
>    at FanControl.ViewModels.AutoFanCurveViewModel.set_SelectedTempSource(ISensorSourceViewModel value)
>    at FanControl.ViewModels.AutoFanCurveViewModel..ctor(IAutoFanCurveConfig autoFanCurve, Func`2 validateName, Action`1 deleteFanCurve, ObservableCollection`1 tempSources, IApplicationClock applicationClock, IDialogService dialogService)
>    at FanControl.ViewModels.AutoFanCurveViewModel.Deserialize(IAutoFanCurveConfig autoFanCurve, Func`2 validateName, Action`1 deleteFanCurve, ObservableCollection`1 tempSources, IApplicationClock applicationClock, IDialogService dialogService)
>    at FanControl.App.<>c__DisplayClass24_0.<RegisterTypes>b__13(MainWindowViewModel c, SerializableAutoFanCurve fc)
>    at FanControl.ViewModels.FanCurveProvider`1.<Deserialize>d__4.MoveNext()
>    at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
>    at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
>    at FanControl.ViewModels.MainWindowViewModel.RestoreApplicationState(SerializableMainWindow state, IFanCurveProvider`1 fanCurveProvider, ICustomSensorProvider`1 customSensorProvider)
>    at FanControl.ViewModels.MainWindowViewModel.RestoreApplicationState(SerializableMainWindow state)
> ```
> 
> Error loading config file, like v122-1 (not sure if exact log message)
